From f840af8058c9a321f58944f17cb39be06719c18d Mon Sep 17 00:00:00 2001
From: Tian Yang <tiany@codeaurora.org>
Date: Thu, 18 Jun 2020 11:44:42 -0700
Subject: [PATCH] skbuff_debug: Fix 64bit ARM issue for kernel 5.4

Kernel 5.4 changes walk_stackframe function argument type in arch/arm64.

Signed-off-by: Tian Yang <tiany@codeaurora.org>
Change-Id: I6f46312d9b419a32c4bebc61321ac0657fec6ed1
---
 net/core/skbuff_debug.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

--- a/net/core/skbuff_debug.c
+++ b/net/core/skbuff_debug.c
@@ -80,8 +80,11 @@ static void skbuff_debugobj_get_stack(vo
 
 	frame.pc = (unsigned long)skbuff_debugobj_get_stack;
 
+#ifdef CONFIG_ARM64
+	walk_stackframe(current,&frame, skbuff_debugobj_walkstack, p);
+#else
 	walk_stackframe(&frame, skbuff_debugobj_walkstack, p);
-
+#endif
 	ret[w.pos] = NULL;
 }
 #else
@@ -272,7 +275,6 @@ static int __init disable_object_debug(c
 early_param("no_skbuff_debug_objects", disable_object_debug);
 
 void skbuff_debugobj_print_skb_list(const struct sk_buff *skb_list,
-	pr_emerg("skb_debug: start skb list '%s'\n", list_title);
 				    const char *list_title, int cpu)
 {
 	int count;
